### ISSUE TITLE
Testnet panicked

### ISSUE THREAD

**Issue by friedger** [blockstack/stacks-blockchain#1318]
I followed the instructions using my own contract on a testnet build from `master` branch. After a while the testnet stops with the following log. To me, it is not clear whether my transaction is invalid or something else is wrong. How can I keep my testnet running?

```
DEBUG [1583918621.940] [src/chainstate/stacks/index/marf.rs:228] Extend b6c055c43ffa9d9dbdd1487859205da30f7b0c8e588975a5efb0edc6817bf8e2 to af425f228a92ebe4d7741b129bb2c2f4326179f682da305b030250ccea9d4cd5
DEBUG [1583918621.945] [src/chainstate/stacks/db/transactions.rs:488] Process transaction eccd14303deeb1f6235624ba18e054790a5bd8c512b0fb6410df95f0ff31a1d1
WARN [1583918621.948] [src/chainstate/stacks/db/blocks.rs:2435] Block a0f6a93e9a37e0fef9375aec3e56021ee32770213b92cee21ddc05e9aa52af71 state root mismatch: expected 1b238b62639f74df22526ce51b4ffbcaf1fa6bd241a62a91496dfd9dc8a1aacd, got 088f115b786367f02f4edfff98c4568b046c512bcf8db9ec17dc2f2f1ffa465d
DEBUG [1583918621.948] [src/vm/clarity.rs:169] Rollback Clarity datastore
DEBUG [1583918621.949] [src/chainstate/stacks/db/blocks.rs:1289] Reject block 54bb1640d3a827800fb042c6676a06962f94eb72e2ef31ccd32326d870797d4b/a0f6a93e9a37e0fef9375aec3e56021ee32770213b92cee21ddc05e9aa52af71
thread 'main' panicked at 'Error while processing block - InvalidStacksBlock("Block a0f6a93e9a37e0fef9375aec3e56021ee32770213b92cee21ddc05e9aa52af71 state root mismatch: expected 1b238b62639f74df22526ce51b4ffbcaf1fa6bd241a62a91496dfd9dc8a1aacd, got 088f115b786367f02f4edfff98c4568b046c512bcf8db9ec17dc2f2f1ffa465d")', src/testnet/node.rs:311:31
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:84
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:61
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1025
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1426
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:65
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:50
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:193
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:210
  10: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:471
  11: rust_begin_unwind
             at src/libstd/panicking.rs:375
  12: std::panicking::begin_panic_fmt
             at src/libstd/panicking.rs:326
  13: blockstack_lib::testnet::node::Node::process_tenure
             at src/testnet/node.rs:311
  14: blockstack_lib::testnet::run_loop::RunLoop::start
             at src/testnet/run_loop.rs:182
  15: blockstack_core::main
             at src/main.rs:215
  16: std::rt::lang_start::{{closure}}
             at /rustc/f3e1a954d2ead4e2fc197c7da7d71e6c61bad196/src/libstd/rt.rs:67
  17: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:52
  18: std::panicking::try::do_call
             at src/libstd/panicking.rs:292
  19: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:78
  20: std::panicking::try
             at src/libstd/panicking.rs:270
  21: std::panic::catch_unwind
             at src/libstd/panic.rs:394
  22: std::rt::lang_start_internal
             at src/libstd/rt.rs:51
  23: std::rt::lang_start
             at /rustc/f3e1a954d2ead4e2fc197c7da7d71e6c61bad196/src/libstd/rt.rs:67
  24: main
  25: __libc_start_main
  26: _start

```

Using the smart contract in the readme, the new testnet still stops. This time with this:
```
DEBUG [1583919790.702] [src/chainstate/stacks/db/blocks.rs:2500] No staging blocks
DEBUG [1583919790.702] [src/chainstate/stacks/db/blocks.rs:2686] No more staging blocks -- processed 0 in total
thread 'main' panicked at 'attempt to add with overflow', src/testnet/run_loop.rs:202:13
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:84
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:61
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1025
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1426
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:65
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:50
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:193
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:210
  10: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:471
  11: rust_begin_unwind
             at src/libstd/panicking.rs:375
  12: core::panicking::panic_fmt
             at src/libcore/panicking.rs:84
  13: core::panicking::panic
             at src/libcore/panicking.rs:51
  14: blockstack_lib::testnet::run_loop::RunLoop::start
             at src/testnet/run_loop.rs:202
  15: blockstack_core::main
             at src/main.rs:215
  16: std::rt::lang_start::{{closure}}
             at /rustc/f3e1a954d2ead4e2fc197c7da7d71e6c61bad196/src/libstd/rt.rs:67
  17: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:52
  18: std::panicking::try::do_call
             at src/libstd/panicking.rs:292
  19: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:78
  20: std::panicking::try
             at src/libstd/panicking.rs:270
  21: std::panic::catch_unwind
             at src/libstd/panic.rs:394
  22: std::rt::lang_start_internal
             at src/libstd/rt.rs:51
  23: std::rt::lang_start
             at /rustc/f3e1a954d2ead4e2fc197c7da7d71e6c61bad196/src/libstd/rt.rs:67
  24: main
  25: __libc_start_main
  26: _start
```

**Comment by jcnelson** [comment]
cc @lgalabru

**Comment by lgalabru** [comment]
Thank you for your feedback @friedger.
These issues are known and fixed awith my ongoing branch, I'll be opening a PR today.

**Comment by whoabuddy** [comment]
I can confirm a similar issue, the same panic happened to me when I first walked through the instructions with the smart contract in the readme (and I assumed it was something I did).

Then twice again today with nothing but the testnet running - no transactions at all.

`thread 'main' panicked at 'attempt to add with overflow', src/testnet/run_loop.rs:202:13`

This last run I timed it, seems to be after ~10min: `real 9m52.147s`

@lgalabru Looking forward to the PR!
